lib/ansiblelint/errors.py

```diff
@@ -18,20 +22,52 @@ def __repr__(self):
         return formatstr.format(self.rule.id, self.message,
                                 self.filename, self.linenumber, self.line)
 
+    def __lt__(self, other):
+        """Enable sorting of MatchError instances."""
+        return (
+                self.filename, self.linenumber, self.message, getattr(self.rule, "id", 0)
+            ) < (
+            other.filename, other.linenumber, other.message, getattr(other.rule, "id", 0))
```

Comment:
I'd put this into a property like `_unique_key` (the actual name could be better), then it'd be way easier to reason about:
```suggestion
        return self._unique_key < other._unique_key

    @property
    def _unique_key(self):
        return (
                self.filename, self.linenumber, self.message, getattr(self.rule, "id", 0)
            )
```